src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -196,6 +196,8 @@ public void setDefinition(FlowDefinition definition) {
         } else {
             quietPeriod = null;
         }
+        getTriggersJobProperty().stopTriggers();
+        getTriggersJobProperty().startTriggers(Items.currentlyUpdatingByXml());
```

Comment:
Not a clue. The trigger stop/start stuff is black magic - I basically just worry about whether the tests pass.